### PR TITLE
Beginner Guide Submission

### PR DESCRIPTION
**Pull request by puneeth072003**
### Submission of pictorial Beginner Guide including "How to add Challenges" part as requested and mentioned in the issue #558 and continuation to the PR #589 

What kind of changes does this PR include?

-   [ ] Fixes or refactors
-   [ ] A new challenge
-   [x] Additional documentation
-   [ ] Something else

Checklist:

-   [ ] All the contributions made are solely the work of me and my co-authors
-   [ ] I tested the changes in this PR (if applicable)
-   [ ] I added unit tests to ensure my change works (when change in Java or on front-end code)
-   [x] The PR passes pre-commit hooks and automated tests

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -99,3 +104,239 @@ Although we greatly appreciate any and all contributions to the project, there a
 -   Unnecessary self-promotion of tools or blog posts is frowned upon. If you have a relation with on of the URLs or tools you are referencing, please state so in the PR so that we can verify that the reference is in line with the rest of the guide.
 
 Please be sure to take a careful look at our [Code of Conduct](https://github.com/OWASP/wrongsecrets/blob/master/CODE_OF_CONDUCT.md) for all the details.
+
+---
+# Beginner guide
+## OWASP WrongSecrets
+
+Basically, [*WrongSecrets*](https://owasp.org/www-project-wrongsecrets/) is an application which teaches how to not store secrets by offering challenges to the user, which helps the user to Self-reflect and correct those mistakes.
```
Comment (bendehaan) [inline CONTRIBUTING.md:112]:
```suggestion
[*WrongSecrets*](https://owasp.org/www-project-wrongsecrets/) is an application teaching how to *not* store secrets by offering challenges to the user, helping the user to Self-reflect and correct those mistakes.
```

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -99,3 +104,239 @@ Although we greatly appreciate any and all contributions to the project, there a
...
+    -  Select ***Edit configuration templates*** then select ***Application*** section.
+    
+       ![](images/edit-config-7.1.png)
+    - There under the ***Application*** section click on the button shown below.
+    
+      ![](images/modify-options-7.2.png)
+    - ***Select*** all the fields that are Selected in the below picture.
+    
+      ![](images/select-options-7.3.png)
+    - ***Fill*** all the fields as shown below.
```
Comment (bendehaan) [inline CONTRIBUTING.md:197]:
```suggestion
    - ***Fill out*** all the fields as shown below.
```

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -99,3 +104,239 @@ Although we greatly appreciate any and all contributions to the project, there a
...
+      ![](images/final-output-8.png)
+
+    - Here is the preview of the **web server**,  you can try to find the secrets by means of solving the challenge offered at:
+      [**Challenges**](https://github.com/OWASP/wrongsecrets#basic-docker-exercises)
+        ![](images/screenshot.png)
+
+---
+## How to add challenge
+- ### Step 1: Creating a new issue.
+  First make sure that you have an [Issue](https://github.com/OWASP/wrongsecrets/issues/new) reported for which a challenge is really wanted, And make sure the challenge is assigned to you, as others might be working at the challenge.
```
Comment (bendehaan) [inline CONTRIBUTING.md:216]:
```suggestion
  First make sure that you have an [Issue](https://github.com/OWASP/wrongsecrets/issues/new) reported for which a challenge is really wanted, And make sure the challenge is assigned to you, as others might be working on the challenge.
```

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -99,3 +104,239 @@ Although we greatly appreciate any and all contributions to the project, there a
...
+            private ScoreCard scoreCard;
+            @Test
+            void rightAnswerShouldSolveChallenge() {
+                var challenge = new Challenge28(scoreCard);
+                Assertions.assertThat(challenge.solved("wrong answer")).isFalse();
+                Assertions.assertThat(challenge.solved(challenge.spoiler().solution())).isTrue();
+            }
+        }
+        ``` 
+  Please note that PRs for new challenges are only excepted when unit tests are added to prove that the challenge works. Normally tests should not immediately leak the actual secret, so leverage the `.spoil()` functionality of your test implementation for this.
```
Comment (bendehaan) [inline CONTRIBUTING.md:307]:
```suggestion
  Please note that PRs for new challenges are only accepted when unit tests are added to prove that the challenge works. Normally tests should not immediately leak the actual secret, so leverage the `.spoil()` functionality of your test implementation for this.
```

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -99,3 +104,239 @@ Although we greatly appreciate any and all contributions to the project, there a
...
+
+- ### Step 4: Adding explanations, reasons and hints.
+  Add the explanation for your challenge along with the hints that will help in finding the secret in this folder `wrongsecrets/src/main/resources/explanations/`.
+  Things to be noted.
+    - All the possible explanations for your challenge, included with all the hints and reasons should be provided.
+    - Every thing must be in a separate **AsciiDoc files**.
```
Comment (bendehaan) [inline CONTRIBUTING.md:313]:
```suggestion
    - Everything must be in separate **AsciiDoc files**.
```

**Review by bendehaan — CHANGES_REQUESTED** [review]
Great work @puneeth072003! A few minor nits and we're good to go. You can just commit these in a batch in the GitHub UI, you don't have to open a new PR.

**Comment by puneeth072003** [comment]
@bendehaan @commjoen If i run `pre-commit run -a` before adding a commit it modifies all the markdown files in the project,
including the Contributing.md shall i commit all the modifications it made?

**Comment by commjoen** [comment]
> @bendehaan @commjoen If i run `pre-commit run -a` before adding a commit it modifies all the markdown files in the project,
> 
> including the Contributing.md shall i commit all the modifications it made?
> 
> In order to pass the pre-commit test....:-)

Yes

**Comment by puneeth072003** [comment]
@bendehaan How can i fix that test that has failed?

**Review by bendehaan — APPROVED** [review]
Looks good now - thanks for your contribution!

**Review by commjoen — APPROVED** [review]
That was a long journey! Awesome! Welcome to the team of contributors!

**Pull request merged**